schema/encoding/jsonschema/jsonschema.go

```diff
@@ -166,29 +168,32 @@ func schemaToJSONSchema(w io.Writer, s *schema.Schema) (err error) {
 		}
 	}
 	ew.writeString("}")
-	ew.writeFormat(`, "required": [%s]`, strings.Join(required, ", "))
+
+	if len(required) > 0 {
+		ew.writeFormat(`, "required": [%s]`, strings.Join(required, ", "))
+	}
 	return ew.err
 }
 
-// Encoder encodes schema.Schema into a JSONSchema
+// Encoder writes the JSON Schema representation of a schema.Schema to an output stream. Note that only a sub-set of the
```

Comment:
Broke the lines at 120 chars. @rs, is there any particular line width you prefer or use elsewhere in this project?